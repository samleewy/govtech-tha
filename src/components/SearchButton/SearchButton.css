button {
    width: 120px;

    background-color: #1C76D5;
    outline: none;
    color: white;
    font-weight: 600;
    border-radius: 8px;
    padding: 11px 20px;

    cursor:pointer;

    /* Remove ugly default stylings */
    outline-style: none;
    box-shadow: none;
    border-color: transparent;
}

button:focus {
    outline: none;
}

i.fas.fa-search {
    margin-right: 5px;
}

@media only screen and (max-width: 600px) {
    .search-btn-text {
        display: none;
    }

    button {
        width: 50px;
        padding:11px 0px;
    }

    i.fas.fa-search { 
        margin-right: 0;
    }
}