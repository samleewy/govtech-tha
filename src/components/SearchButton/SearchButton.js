import React from 'react'
import './SearchButton.css'

const SearchButton = ({ onSearchClick }) => {

    return (
        <button 
            onClick={onSearchClick}>
            <i className="fas fa-search"></i> Search
        </button>
    )
}

export default SearchButton