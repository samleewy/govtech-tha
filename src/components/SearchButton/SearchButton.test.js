import { render } from '@testing-library/react';
import SearchButton from './SearchButton'

describe("This suit is to test SearchButton component", () => {
    it('should render correctly', () => {
        const { asFragment } = render(<SearchButton />)
        expect(asFragment()).toMatchSnapshot();
    })
})