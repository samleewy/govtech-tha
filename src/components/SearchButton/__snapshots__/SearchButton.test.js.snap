// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test SearchButton component should render correctly 1`] = `
<DocumentFragment>
  <button>
    <i
      class="fas fa-search"
    />
    <span
      class="search-btn-text"
    >
      Search
    </span>
  </button>
</DocumentFragment>
`;
